### PR TITLE
Added content specific links for StarkScan block explorer

### PR DESCRIPTION
**Pull request by milancermak**
## Introduced changes

- Links to StarkScan are now differentiated by type. Instead of the generic `/search/<HASH>`, they go to `/tx/<HASH>`, `/class/<HASH>` or `/contract/<ADDR>`, similar as with Voyager.

## Checklist

- [ ] Linked relevant issue
- [x] Updated relevant documentation
- [ ] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Review by cptartur — APPROVED** [review]

**Review by integraledelebesgue — CHANGES_REQUESTED** [review]
StarkScan interaction is going to change a little bit [here](https://github.com/foundry-rs/starknet-foundry/pull/2466#issue-2522247149). Please wait until this PR gets merged and update your fix then.

**Comment by milancermak** [comment]
@integraledelebesgue done

**Review by integraledelebesgue — APPROVED** [review]

**Pull request merged**
